File: src/main/java/org/kite9/diagram/dom/model/DiagramElementFactory.java
package org.kite9.diagram.dom.model;

import org.kite9.diagram.dom.elements.Kite9XMLElement;
import org.kite9.diagram.model.DiagramElement;

/**
 * Allows for the conversion from an `XMLElement` to a Kite9 `DiagramElement` (used in layout).
 * 
 * @author robmoffat
 *
 */
public interface DiagramElementFactory {

	public DiagramElement createDiagramElement(Kite9XMLElement in, DiagramElement parent);
	
}

File: kite9-visualization/src/common/kotlin/org/kite9/diagram/common/elements/factory/DiagramElementFactory.kt
package org.kite9.diagram.common.elements.factory

import org.kite9.diagram.dom.bridge.ElementContext
import org.kite9.diagram.model.DiagramElement

/**
 * Allows for the conversion from an `X` (usually XML element) to a Kite9 `DiagramElement` (used in layout).
 *
 * @author robmoffat
 */
interface DiagramElementFactory<X> {

    fun setElementContext(ec: ElementContext)

    fun createDiagramElement(x: X, parent: DiagramElement?): DiagramElement?

    fun createTemporaryConnected(parent: DiagramElement, idSuffix: String): TemporaryConnectedRectangular

}